js/components/Compass/index.js

```diff
@@ -0,0 +1,110 @@
+import React, { Component } from 'react';
+import {
+  Image,
+  Animated,
+  DeviceEventEmitter
+} from 'react-native';
+import ReactNativeHeading from 'react-native-heading';
+
+export default class Compass extends Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      defaultAzimut: props.azimut || 360,
+      rotation: new Animated.Value(0),
+      headingIsSupported: false,
+      angleFilter: props.degreesFilter || 2
+    }
+  }
+
+  setFilter(filter) {
+    this.state.angleFilter = parseInt(filter);
+    this.stopHeadingUpdates();
+    this.startHeadingUpdates();
+  }
+
+  startHeadingUpdates() {
+    ReactNativeHeading.start(this.state.angleFilter)
```

Comment:
It doesn't matter, same than gps is a sensor that is being already used, and we will listen to the updated values.